app/models/server/raw/LivechatRooms.js

```diff
@@ -1,5 +1,81 @@
 import { BaseRaw } from './BaseRaw';
+import { getValue } from '../../../settings/server/raw';
 
 export class LivechatRoomsRaw extends BaseRaw {
-
+	async findAllNumberOfAbandonedRooms({ start, end, departmentId, options = {} }) {
+		const roomsFilter = [
```

Comment:
Why are you applying these filters inside the `$project` instead of the `$match` statement?
Using this approach you will, first, return all rooms with the `t: 'l'` into the `$project` and then you will apply the rest of the filters..
Also, since the room is considered abandoned after closing it, you should filter the `room.closedAt` field instead of the `room.ts`, right?